src/net_processing.cpp

```diff
@@ -1961,7 +1965,10 @@ void PeerManagerImpl::BlockDisconnected(const std::shared_ptr<const CBlock> &blo
  * Maintain state about the best-seen block and fast-announce a compact block
  * to compatible peers.
  */
-void PeerManagerImpl::NewPoWValidBlock(const CBlockIndex *pindex, const std::shared_ptr<const CBlock>& pblock)
+void PeerManagerImpl::NewPoWValidBlock(
```

Comment:
d3b533d3f1e80addca2ab0b3ee97215d5d03d984

could add an assert for the invariant: can't be BACKGROUND role.

Reply:
This is done later infbd05a9eed1e40be0c4915e2aef98b3771d7af6c, although in a `return` way for some reason?